### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by petro2061**
This task added classes to test the services, repositories, and controllers for the book and category entities. To test the service level, the mock-up object approach was used. Testing of the storage layer takes place in test containers to simulate real databases. The controller level is tested using integration tests.
test-with-coverage

### CODE REVIEW

File: src/main/java/project/onlinebookstore/service/book/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -31,6 +31,7 @@ public BookDto save(CreateBookRequestDto bookRequestDto) {
         return bookMapper.toBookDto(book);
     }
 
+    @Transactional
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/service/book/impl/BookServiceImpl.java]: why did you add this annotation?

File: src/main/java/project/onlinebookstore/service/book/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -39,6 +40,7 @@ public List<BookDto> findAll(Pageable pageable) {
                 .toList();
     }
 
+    @Transactional
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/service/book/impl/BookServiceImpl.java]: why did you add this annotation?

File: src/test/java/project/onlinebookstore/controller/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+        MvcResult result = mockMvc.perform(get("/books")
+                        .param("page", "0")
+                        .param("size", "10")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+        //Then
+        byte[] jsonResponse = result.getResponse().getContentAsByteArray();
+        BookDto[] bookDtoResponseArray = objectMapper.readValue(jsonResponse, BookDto[].class);
+        List<BookDto> actualBookDtoList = Arrays.stream(bookDtoResponseArray).toList();
```
Comment (Elena-Bruyako) [inline src/test/java/project/onlinebookstore/controller/BookControllerTest.java]:
```suggestion
        List<BookDto> actualBookDtoList = objectMapper.readValue(result.getResponse().getContentAsString(),
                new TypeReference<>() {});
```

File: src/test/java/project/onlinebookstore/controller/BookControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+        Assertions.assertFalse(actualBookDtoList.isEmpty());
```
Comment (Elena-Bruyako) [inline src/test/java/project/onlinebookstore/controller/BookControllerTest.java]: use static imports, check all places

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Comment by petro2061** [comment]
Removed Transactional annotation added before. Also removed reduant properties in application.properties file for tests. Added static imports in test classes.

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**